routes/clarivate.js

```diff
@@ -208,33 +189,63 @@ function getAdmin(req, res, uri, callback) {
         callback);
 };
 
-function getFilteredConsumerId(req,res) {
-  getAdmin(req, res, '/consumers/'+req.query.consumerid, (err, consumer) => {
-      if (err) {
-          return next(err);
-      }
-  let body = utils.getJson(consumer.body);
-      if (body && body.message === "Not found") {
-          res.json({
-              title: 'All Subsriptions',
-              subscriptions: null
-          });
-      } else {
-          //Desctruct body.username(for example app1$mockbin) and assign to variables
-          let [appId, apiId] = (body.username).split("$");
-          utils.getFromAsync(req, res, `/applications/${appId}/subscriptions/${apiId}`, 200, (err, application) => {
-              if (err) {
-                  return next(err);
-              }
-              application.consumerid = req.query.consumerid;
-      let subscriptions = {items: [application],count: 1}
-              res.json({
-                  title: 'All Subsriptions',
-                  subscriptions
-              });
-          })
-      }
-  });
+function getSubscriptions(req, res,) {
+	const filterFields = ['application', 'plan', 'api'];
+	const subsUri = utils.makePagingUri(req, '/subscriptions?embed=1&', filterFields);
+	utils.getFromAsync(req, res, subsUri, 200, function (err, subsResponse) {
+		if (err) {
+			return next(err);
+		} 
+		let consumerIds = [];
+		for (let i = 0; i < subsResponse.items.length; ++i) {
+			let consumerUsername = `${subsResponse.items[i].application}$${subsResponse.items[i].api}`; 
+			consumerIds.push(consumerUsername)
+		}
+		async.map(consumerIds, function (consumerid, callback) {
+				getAdmin(req, res, '/consumers/' + consumerid, callback);
+		}, function (err, consumersResponse) {
+			consumersResponse.map((consumer, i) => {
+				let body = utils.getJson(consumer.body);
+				if (consumerIds[i] === body.username) {
+					subsResponse.items[i].consumerid = body.id;
+				} 
+			})
+			res.json({
+				title: 'All Subsriptions',
+				subscriptions: subsResponse
+			});
+		});
+	});
+}
+
+function getFilteredConsumerId(req,res, callback) {
+    getAdmin(req, res, '/consumers/'+req.query.consumerid, (err, consumer) => {
+        if (err) {
+            return next(err);
+		}
+		let body = utils.getJson(consumer.body);
+        if (body && body.message === "Not found") {
+            res.json({
+                title: 'All Subsriptions',
+                subscriptions: null
+            });
+        } else {
+            //Desctruct body.username(for example app1$mockbin) and assign to variables
+            let [appId, apiId] = (body.username).split("$");
+            utils.getFromAsync(req, res, `/applications/${appId}/subscriptions/${apiId}`, 200, (err, application) => {
```

Comment:
I thought this is the call we are trying to avoid. This is the call which will not work for approver. This function should only return back appId and apiId

Reply:
Yes, got it. This code is also working for approvers, but there is a lot of extra code